algorithms/search/jump_search.py

```diff
@@ -0,0 +1,38 @@
+import math
+
+def jump_search(arr,target):
+    '''Jump Search
+        Worst-case Complexity: O(√n) (root(n))
+        All items in list must be sorted like binary search
+
+        Find block that contains target value and search it linearly in that block
+
+        reference: https://en.wikipedia.org/wiki/Jump_search
+
+    '''
+    n = len(arr)
+    block_size = int(math.sqrt(n))
+    block_prev = 0
+    block= block_size
+
+    #return -1 means that array doesn't contain taget value
+    #find block that contains target value
+    while arr[min(block,n)-1] < target :
+        block_prev = block
+        block += block_size
+        if block_prev >=n :
+            return -1
```

Comment:
The index of `arr` in the `while` condition looks cluttered, here, a better piece of code can be this. ( i hope it doesn't has some bug, feel free to point-out)
```python
if arr[n - 1] > target:
    return -1  # array doesn't contain target value
while block <= n and arr[block - 1] < target:
    block_prev = block
    block += block_size
```